photos/user.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2016 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+// Author: Spencer Kimball (spencer.kimball@gmail.com)
+
+package main
+
+import (
+	"database/sql"
+	"math"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/cockroachdb/cockroach/util"
+	"github.com/cockroachdb/cockroach/util/log"
+	"github.com/cockroachdb/cockroach/util/stop"
+)
+
+const (
+	meanUserID    = 1 << 15
+	rate          = 1.0 / (meanUserID * 2)
+	statsInterval = 10 * time.Second
+)
+
+const (
+	createUserOp = iota
+	createPhotoOp
+	createCommentOp
+	listPhotosOp
+	listCommentsOp
+	updatePhotoOp
+	updateCommentOp
+	deletePhotoOp
+	deleteCommentOp
+	lastOp
+)
+
+type opDesc struct {
+	typ      int
+	name     string
+	relFreq  float64
+	normFreq float64
+}
+
+var ops = []*opDesc{
+	{createUserOp, "create user", 1, 0},
+	{createPhotoOp, "create photo", 10, 0},
+	{createCommentOp, "create comment", 50, 0},
+	{listPhotosOp, "list photos", 20, 0},
+	{listCommentsOp, "list comments", 20, 0},
+	{updatePhotoOp, "update photo", 2, 0},
+	{updateCommentOp, "update comment", 5, 0},
+	{deletePhotoOp, "delete photo", 1, 0},
+	{deleteCommentOp, "delete comment", 2.5, 0},
+}
+
+var mu sync.Mutex
+var computing = false
+var totalOps int
+var maxUserID int
+var noUserOps int
+var failedOps int
+var opCounts = map[int]int{}
+var histogram = map[int]int{}
```

Comment:
If that `mu` protects all these fields, you can make it more obvious by structuring it like:

```
var stats struct {
  sync.Mutex
  computing bool
  totalOps int
  maxUserID int
  ...
}
```

And then when you use it you do:

```
  stats.Lock()
  defer stats.Unlock()
  if stats.computing {
    return
  }
```